Update Reverse string to new format (#2363)
exercism/problem-specifications: exercises/reverse-string/instructions.md

# Instructions

Your task is to reverse a given string.

Some examples:

- Turn `"stressed"` into `"desserts"`.
- Turn `"strops"` into `"sports"`.
- Turn `"racecar"` into `"racecar"`.

> Many languages provide built-in functions or methods for reversing strings, but for this exercise, presume you have to write it from scratch.

Reviewer: ```suggestion
Many languages provide built-in functions or methods for reversing strings; for this exercise, write your own from scratch.
```

> Many languages provide built-in functions or methods for reversing strings, but for this exercise, presume you have to write it from scratch.

Reviewer 2: I dont know if this kind of input should be included, for the first so does the majority of solutions use some form of built-in methods to reverse the string. It is also very vividly written and could mean a little bit of everything.

Like, here is "clearly" a method used:

```ruby
text = "ABC"
text.reverse
# => "CBA"
```

But here you could argue you are not using a "reverse" method/function but could still be seen as it:

```python
>>> text = "ABC"
>>> text[::-1]
"CBA"
```

And is an approach like, here I don't use any methods on strings to reverse it but I convert it to another datatype , is that categorized as "built-in" methods, or is the fact that I use reverse on an array seen as I use "built-in" methods?

```ruby
text = "ABC"
characters = text.split
characters.reverse
characters.join
# => "CBA"
```

> 
> Many languages provide built-in functions or methods for reversing strings, but for this exercise, presume you have to write it from scratch.

Author: ```suggestion
```
